src/System.CommandLine/Builder/CommandLineBuilderExtensions.cs

```diff
@@ -562,17 +482,12 @@ public static CommandLineBuilder UseHelp(this CommandLineBuilder builder, int? m
             this CommandLineBuilder builder,
             int maxLevenshteinDistance = 3)
         {
-            builder.AddMiddleware(async (context, next) =>
+            if (maxLevenshteinDistance <= 0)
```

Comment:
we should throw immediately rather than when doing first correction:

https://github.com/dotnet/command-line-api/blob/7030070c394c2d85dd10f6d1cc22ac6c1e902ce2/src/System.CommandLine/Invocation/TypoCorrection.cs#L14-L19

Reply:
Let's add it to a list of braking changes related to configuration and include in one PR with config re-design (somewhen next week?)